### PR TITLE
Make websockets chat service use the new chat service base classes

### PR DESCRIPTION
**Pull request by ideanl**
**Patch description**
This PR changes the websockets chat service to use the newly made base classes for chat services. The message format has been changed to conform to the base class requirement of having a dict with a sender key, which is now the case. `_parse_config` and `_complete_setup` are no longer being called from init in the base chat service manager vs. the messenger implementation, so I had to call them when overriding init.

**Testing steps**
See testing steps in #2132

### CODE REVIEW

File: parlai/core/params.py [hunk 1]
```diff
@@ -458,6 +458,13 @@ def add_websockets_args(self):
         websockets.add_argument(
             '--port', default=35496, type=int, help='Port to run the websocket handler'
         )
+        websockets.add_argument(
+            '--password',
```
Comment (klshuster) [inline parlai/core/params.py]: i'm not sure if it's there yet, but this should be eventually wrapped up in some `add_chat_service_args` function

File: parlai/chat_service/services/websocket/websocket_manager.py [hunk 2]
```diff
@@ -9,234 +9,68 @@
...
+        self.agent_id_to_overworld_future = {}
+        self.task_to_agent_ids = {}
         self._load_model()
 
     def _load_model(self):
         """Load model if necessary"""
         if 'model_file' in self.opt or 'model' in self.opt:
             self.opt['shared_bot_params'] = create_agent(self.opt).share()
 
+    def _handle_message_read(self, event):
```
Comment (emilydinan) [inline parlai/chat_service/services/websocket/websocket_manager.py]: what is this for? can you add a docstring?

File: parlai/chat_service/services/websocket/websocket_manager.py [hunk 3]
```diff
@@ -418,7 +191,7 @@ def _create_agent(self, task_id, socketID):
         :param agent_id:
             int agent id
         """
-        return WebsocketAgent(self.opt, self, task_id, socketID)
+        return WebsocketAgent(self.opt, self, socketID, task_id)
```
Comment (emilydinan) [inline parlai/chat_service/services/websocket/websocket_manager.py]: ooo tricky that we swapped this

File: parlai/chat_service/services/websocket/websocket_manager.py [hunk 4]
```diff
@@ -451,6 +224,17 @@ def observe_message(self, socket_id, text):
         asyncio.set_event_loop(asyncio.new_event_loop())
         return MessageSocketHandler.subs[socket_id].write_message(text)
 
-    def _log_debug(self, text):
-        time = datetime.datetime.now().strftime('%Y-%m-%d %H:%M:%S')
-        shared_utils.print_and_log(logging.DEBUG, f'{time}: {text}', should_print=True)
+    def restructure_message():
```
Comment (emilydinan) [inline parlai/chat_service/services/websocket/websocket_manager.py]: can we add docstrings here too?

File: parlai/chat_service/services/websocket/websocket_manager.py (hunk 2 above)
Comment (ideanl) [reply parlai/chat_service/services/websocket/websocket_manager.py]: Added a docstring

File: parlai/chat_service/services/websocket/websocket_manager.py (hunk 3 above)
Comment (ideanl) [reply parlai/chat_service/services/websocket/websocket_manager.py]: Yeah I had it in the wrong order before the base class was created. Now it's consistent with the other agents

File: parlai/chat_service/services/websocket/websocket_manager.py (hunk 4 above)
Comment (ideanl) [reply parlai/chat_service/services/websocket/websocket_manager.py]: Added a docstring

File: parlai/core/params.py (hunk 1 above)
Comment (ideanl) [reply parlai/core/params.py]: Created the function and put common args between the services in it

**Pull request merged**